patches/015-fix-linkage.patch

```diff
@@ -0,0 +1,25 @@
+From 26cd1bf0999e4174a2f0ac46692370d118b98d67 Mon Sep 17 00:00:00 2001
+From: Simran Singh <simran.p.singh@oracle.com>
+Date: Tue, 14 May 2024 16:23:46 +0000
+Subject: [PATCH] empty toolenv
+
+---
+ src/cmd/dist/build.go | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
+index 78ebec404b..58398a7fbb 100644
+--- a/src/cmd/dist/build.go
++++ b/src/cmd/dist/build.go
+@@ -1315,7 +1315,7 @@ func toolenv() []string {
+ 		// we disable cgo to get static binaries for cmd/go and cmd/pprof,
+ 		// so that they work on systems without the same dynamic libraries
+ 		// as the original build system.
+-		env = append(env, "CGO_ENABLED=1")
++		env = append(env, "")
```

Comment:
yes, but removing the entire conditional branch kind of takes away the original source code's flow. Also having the rest of comments in there helps preserve the original author's intent and makes leeway for future changes - e.g. bringing CGO_ENABLED back. No ?

Reply:
@dbenoit17 let me know how you want me to proceed. I would really like this patch in asap. So i can replicate the centos patch if u prefer that.

Reply:
I would say, if we wanted to preserve some of the original intent, we could comment out the `env = append(env, "CGO_ENABLED=0")` instead of replacing it with an empty string append, and add an additional comment explaining why we're making this change. Would you mind adding that in @sipasing ? Do you agree with this @dbenoit17 ? That actually makes it clearer for anyone reading the code. The compiler should eliminate the branch anyways, so no worries there.

Reply:
@derekparker that sounds good to me

Reply:
How does it look now ?

Reply:
@dbenoit17 Can you let me know if the comment looks good now ,and then I can change #194 also

Reply:
lgtm, thanks!